classes/security/Validation.php

```diff
@@ -474,6 +482,83 @@ public static function canAdminister($administeredUserId, $administratorUserId)
         return true;
     }
+
+    /**
+     * Get the user's administration level
+     *
+     * @param int   $administeredUserId     User ID of user to potentially administer
+     * @param int   $administratorUserId    User ID of user who wants to do the administrating
+     * @param int   $contextId              The journal/context Id
+     * 
+     * @return int The authorized adminstration level 
+     */
+    public static function getAdministrationLevel(int $administeredUserId, int $administratorUserId, int $contextId = null): int
+    {
+        // You can administer yourself
+        if ($administeredUserId == $administratorUserId) {
+            return self::ADMINISTRATION_FULL;
+        }
+
+        $filteredSiteAdminUserGroups = Repo::userGroup()
+            ->getCollector()
+            ->filterByContextIds([\PKP\core\PKPApplication::CONTEXT_SITE])
+            ->filterByRoleIds([Role::ROLE_ID_SITE_ADMIN]);
+
+        // You cannot adminster administrators
+        if ($filteredSiteAdminUserGroups->filterByUserIds([$administeredUserId])->getCount() > 0) {
+            return self::ADMINISTRATION_PROHIBITED;
+        }
+
+        // Otherwise, administrators can administer everyone
+        if ($filteredSiteAdminUserGroups->filterByUserIds([$administratorUserId])->getCount() > 0) {
+            return self::ADMINISTRATION_FULL;
+        }
+
+        // Check for administered user group assignments in other contexts
+        // that the administrator user doesn't have a manager role in.
+        $userGroupsCount = Repo::userGroup()
+            ->userUserGroups($administeredUserId)
+            ->filter(fn($userGroup) => 
+                $userGroup->getContextId() != \PKP\core\PKPApplication::CONTEXT_SITE &&
+                !Repo::userGroup()
+                    ->getCollector()
+                    ->filterByContextIds([$userGroup->getContextId()])
+                    ->filterByUserIds([$administratorUserId])
+                    ->filterByRoleIds([Role::ROLE_ID_MANAGER])
+                    ->getCount()
+            )
+            ->count();
+        
+        if ( $userGroupsCount > 0 ) {
+            // Found an assignment: disqualified.
+            // But also determine if a partial administrate is allowed
+            // if the Administrator User is a Journal Manager in the current context
+            if ($contextId && 
+                Repo::userGroup()
+                    ->getCollector()
+                    ->filterByContextIds([$contextId])
+                    ->filterByUserIds([$administratorUserId])
+                    ->filterByRoleIds([Role::ROLE_ID_MANAGER])
+                    ->getCount()) {
+                return self::ADMINISTRATION_PARTIAL;
+            }
+            return self::ADMINISTRATION_PROHIBITED;
+        }
+
+        // Make sure the administering user has a manager role somewhere
+        $roleManagerCount = Repo::userGroup()
```

Comment:
This check should be moved above any potential `ADMINISTRATION_FULL` or `ADMINISTRATION_PARTIAL` return. (You could re-use the query in https://github.com/pkp/pkp-lib/pull/8283/files#r989522297 for this for bonus points!)

Reply:
@asmecher I think so but I think it should not be right at the top but right after following three checks 
1. user can administrate himself/herself.
2. can not administrate an administrator.
3. an administrator can administrate anyone .

updated accordingly .